pandas/core/indexes/datetimes.py

```diff
@@ -298,41 +298,22 @@ def __new__(cls, data=None,
                 tz = timezones.maybe_get_tz(tz)
                 data = conversion.tz_localize_to_utc(data.view('i8'), tz,
                                                      ambiguous=ambiguous)
-            subarr = data.view(_NS_DTYPE)
         else:
             # must be integer dtype otherwise
             # assume this data are epoch timestamps
             if data.dtype != _INT64_DTYPE:
                 data = data.astype(np.int64, copy=False)
-            subarr = data.view(_NS_DTYPE)
 
+        subarr = data.view(_NS_DTYPE)
         assert isinstance(subarr, np.ndarray), type(subarr)
         assert subarr.dtype == 'M8[ns]', subarr.dtype
 
         subarr = cls._simple_new(subarr, name=name, freq=freq, tz=tz)
-        if dtype is not None:
-            if not is_dtype_equal(subarr.dtype, dtype):
```

Comment:
This check is made unnecessary by the `tz = dtl.validate_tz_from_dtype(dtype, tz)` above (implemented a couple PRs ago)

Reply:
Nope, this isn't quite right.  Will revert/fix.